src/server/multi_command_squasher.cc

```diff
@@ -161,22 +165,29 @@ OpStatus MultiCommandSquasher::SquashedHopCb(EngineShard* es, RespVersion resp_v
   DCHECK(!sinfo.dispatched.empty());
 
   auto* local_tx = sinfo.local_tx.get();
-  facade::CapturingReplyBuilder crb(ReplyMode::FULL, resp_v);
+  CapturingReplyBuilder crb(ReplyMode::FULL, resp_v);
   ConnectionContext local_cntx{cntx_, local_tx};
   if (cntx_->conn()) {
     local_cntx.skip_acl_validation = cntx_->conn()->IsPrivileged();
   }
 
   CmdArgVec arg_vec;
 
+  auto move_reply = [&sinfo](CapturingReplyBuilder::Payload&& src,
+                             CapturingReplyBuilder::Payload* dst) {
+    *dst = std::move(src);
+    size_t sz = Size(*dst);
+    sinfo.reply_size_delta += sz;
+    sinfo.reply_size_total_ptr->fetch_add(sz, std::memory_order_relaxed);
```

Comment:
here we update a thread local atomic variable from the shard thread. the differrence is that now we have K atomic variables - one per each I/O thread

Reply:
Is it better to have k atomic variables, one per thread than 1 single atomic?

Reply:
Yes, having a single global atomic  creates a huge bottleneck. You can easily check it by writing a short c++ program that launches K threads that continuously fetch_add a single atomic uint. At the end it requires synchronizing a single memory cell across many CPUs. It's really bad.